src/4_run_evaluation.py

```diff
@@ -333,22 +336,44 @@ def extract_values(self, inputs: List[Input]):
                     f"return Array.from(document.getElementsByName(`{input.name}`)).map((element) => element.value);"
                 )
 
-                # commenting out this asssrtion since there could be more than one text input with the same name.
+                if input.type in ['textarea']:
+                    visible_values = self.driver.execute_script(
+                        f"return Array.from(document.getElementsByName(`{input.name}`)).map((element) => element.innerHTML);"
+                    )
+                elif input.type == 'select':
+                    visible_values = self.driver.execute_script(
+                        f"return Array.from(document.getElementsByName(`{input.name}`)[0].children).filter((el) => el.selected == true).map((el) => el.value);"
```

Comment:
if this works, that's good. for consistency, might want to check to see if we can do getElementsByName[0] for the other ones (above and below)

Reply:
It works. I actually like `querySelector` and `querySelectorAll` a bit more if we want to find something consistent across the board.